azure/services/loadbalancers/client.go

```diff
@@ -18,158 +18,127 @@ package loadbalancers
 
 import (
 	"context"
-	"encoding/json"
 
-	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-08-01/network"
-	"github.com/Azure/go-autorest/autorest"
-	azureautorest "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
 	"github.com/pkg/errors"
-	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/asyncpoller"
 	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
 // azureClient contains the Azure go-sdk Client.
 type azureClient struct {
-	loadbalancers network.LoadBalancersClient
+	loadbalancers *armnetwork.LoadBalancersClient
+	auth          azure.Authorizer
 }
 
 // newClient creates a new load balancer client from subscription ID.
-func newClient(auth azure.Authorizer) *azureClient {
-	c := newLoadBalancersClient(auth.SubscriptionID(), auth.BaseURI(), auth.Authorizer())
-	return &azureClient{c}
-}
+func newClient(auth azure.Authorizer) (*azureClient, error) {
+	opts, err := azure.ARMClientOptions(auth.CloudEnvironment())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get load balancer client options")
+	}
 
-// newLoadbalancersClient creates a new load balancer client from subscription ID.
-func newLoadBalancersClient(subscriptionID string, baseURI string, authorizer autorest.Authorizer) network.LoadBalancersClient {
-	loadBalancersClient := network.NewLoadBalancersClientWithBaseURI(baseURI, subscriptionID)
-	azure.SetAutoRestClientDefaults(&loadBalancersClient.Client, authorizer)
-	return loadBalancersClient
+	factory, err := armnetwork.NewClientFactory(auth.SubscriptionID(), auth.Token(), opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create load balancer client factory")
+	}
+	return &azureClient{factory.NewLoadBalancersClient(), auth}, nil
 }
 
 // Get gets the specified load balancer.
 func (ac *azureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (result interface{}, err error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "loadbalancers.azureClient.Get")
 	defer done()
 
-	return ac.loadbalancers.Get(ctx, spec.ResourceGroupName(), spec.ResourceName(), "")
+	resp, err := ac.loadbalancers.Get(ctx, spec.ResourceGroupName(), spec.ResourceName(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
```

Comment:
```suggestion
	return resp.LoadBalancer, nil
```

This is what needs to be returned, but unfortunately the signature is just `interface{}` so we can't enforce the return value.

Reply:
Updated